### DISCUSSION TITLE
Sync fork stopped working under "Allow fork syncing" branch protection

### DISCUSSION THREAD

**Discussion by crisman** [community/community#57289] [Repositories]
### Select Topic Area

Bug

### Body

Given a repo I own that is a fork of another repo I don't own.

Some time in the last few weeks (on or after May 18) a branch protection rule with these checked:

- Lock branch
  - Allow fork syncing
- Do not allow bypassing the above settings

has stopped allowing `Sync fork` > `Update branch` to work with an error:

    protected branch 'main' check failed: Cannot change this locked branch

I was able to successfully sync a forked repo I have on May 18 and many days before that.

Not working in any of my forks I have tested in the last five days or so.

Reproduction Steps

- Fork a repo from another account
- When prompted, set a branch protection on the main (or whichever branch was forked)
  - Lock branch
    - Allow fork syncing
  - Do not allow bypassing the above settings
- Wait for (or make a commit) in the original repo
- From repo web page for the fork
- Sync fork
  - Update branch
    - Get error "protected branch 'main' check failed: Cannot change this locked branch"

Has anyone else seen this new failure mode?

Does anyone have a sync that has worked to a protected branch more recently than May 18?

**Comment by crisman** [comment]
Other notes:
-  syncs are working fine if I don't have the branch protection set up
- A new fork with a new branch protection set up as above does break sync with the above error

**Comment by rconnell28** [comment]
This is also happening to my repositories as well. I think it started happening after May 19th for me.

**Comment by hzierer** [comment]
We encountered the same in our organization. As a workaround we switched the forked repos to the new (beta) rulesets: https://github.com/ \<org name\>/\<repo name\>/settings/rules and removed the "old" branch protections.

**Comment by tonglil** [comment]
This is an issue as well.
This was a FF git merge according to the docs here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-branch-from-the-command-line

<img width="647" alt="Screenshot 2023-06-09 at 12 58 52 PM" src="https://github.com/community/community/assets/3250776/6a79da8b-b499-4824-beee-46cee01d6498">
ranch-from-the-command-line

**Comment by DNin01** [comment]
The solution is to upgrade to [rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets), under Settings → Rules → Rulesets. Rulesets don't cause this bug.

**Comment by patrick-knight** [comment]
Wanted to follow up that a fix was shipped last week and wanted to confirm that with y'all!

**Reply by TBBle** [reply]
I've confirmed that both my repo using branch protection (and the repo I converted to using rulesets due to this issue) are happily syncing from their fork-parent.

Thank you for the fix.

**Comment by rainbow1016** [comment]
very good question!

**Comment by lubo** [comment]
When _Require a pull request before merging_ is enabled, this doesn't work but no error message is provided.
